### ISSUE TITLE
documentation & use exmaples

### ISSUE THREAD

**Issue by general-albatross** [stuartpittaway/WebastoHeaterWBusArduinoInterface#1]
Hi Stuart,
Do you have any further documentation on wbus protocol used on Webasto and maybe some use exapmles for your library you can share?
Regards,
Thomas

**Comment by stuartpittaway** [comment]
Hello, think you are the first person to ask!

I've not picked up this project for a while, but I did managed to get my
heater to fire up and run.  My goal was to create an Arduino based
controller with an LCD display - this is what the code on GITHUB does.

Most of my code logic comes from reading the topics and ideas on this page

http://sourceforge.net/p/libwbus/discussion/756917/

**Comment by Mogensen76** [comment]
Do you have any schematic?

Br Johnny

**Comment by stuartpittaway** [comment]
Yes, see this project.

https://github.com/stuartpittaway/WebastoWBUSInterface

**Comment by Mogensen76** [comment]
Thank you very much :-)

**Comment by boelle** [comment]
a long time ago... 

but has any tried to add a gsm module to the mix?

**Comment by stuartpittaway** [comment]
No I don't think so

**Comment by boelle** [comment]
well i have the hardware... only need a levelshifter between gsm and arduino (the mega) and also i like a  L9637D to do the conversion between the wbus and arduino

and i need to dig out some sample code for the gsm part

**Comment by stuartpittaway** [comment]
Should be fairly easy to do. Let me know how you get on

**Comment by boelle** [comment]
will do.... 

how would you judge this code? http://pastebin.com/9jEW030e
the wbus.h for that sketch: http://pastebin.com/8e8JQdX0

a bit different from the one you use.... i dont have a clue which one is the better one here 

its from another project that might be better  suited for the task...

but i take any advice i can get

EDIT: https://www.youtube.com/watch?v=6zZjSFef4yA

a demo of it... not my work...

**Comment by boelle** [comment]
my plan was to cut out code that i do not need, then add in gsm stuff

and try to find documentation on the strings that webasto uses for status etc,  with some damn great luck i can use their app to control it :-D

**Comment by stuartpittaway** [comment]
Looks okay, memory size is the biggest problem when using LCD screens and
interfacing to other devices.

**Comment by boelle** [comment]
oki doke

lcd is going out and this will be hidden either under the dash or seat

only need to get some components to do level shifting... the GSM module is on 2.8V so it needs slight adjusting before it will work right

btw... is there a complete list of commands i can send the preheater? just so i can figure if there are any other nice stats i can get out of it or the sketch i found will make do

**Comment by stuartpittaway** [comment]
I've not found a reference manual as such, there is a link in my source
code to the site where I found most information from.

**Comment by boelle** [comment]
oki, wil try and have a look

when i come up with a sketch will you have time to look it over and suggest corrections?

what i'm after is to get all "dead" code out

but it will not be before next month, to busy right now to so much more

**Comment by stuartpittaway** [comment]
Happy to take a look.

**Comment by boelle** [comment]
bored enough that i started to have a hack at the code

http://pastebin.com/kNdYcnyN

this is just for a few hours of work, the lcd stuff is slowly going out to be replaced with texting commands to the arduino and it sending status back

Q: there are 2 get status voids.... are the variables in those accessible from the main loop?

i ask because i cant remember and i want to send the sms/text from the main loop if i can... but if not i will have to do it from the get status void

**Comment by boelle** [comment]
or is the it the other way round.. ie variables in the main loop are available in the get status voids ?

if that is the case no problem... i guess it would make sense to send the sms from there anyway...

if i call the status voids once they are done i will be returned from where i left off in the main void right?

a bit rusty so sorry for asking questions

**Comment by stuartpittaway** [comment]
Yes I think you are correct, not 100% following, but I think you are on the
right lines.

**Comment by boelle** [comment]
where do i loose you?

**Comment by boelle** [comment]
what i mean is that sometimes variables are not "useable" in the main loop... it fails and says that its not declared etc...

my problem here is that when i get the text in the main loop it stores the number of the sender in the number variable... so my plan is to call the get status void... then send the text from within the void and use the number variable to tell who should get the text

**Comment by boelle** [comment]
my sketch so far... have started to mark lines that might be dead code

http://pastebin.com/ie806tp5

will continue tomorrow... any headsup are welcome

**Comment by boelle** [comment]
tried to compile and it works except one line i commented out

http://pastebin.com/zP6qdPNp

any recommendations? did i get {} right in the main loop?

my logic for the main loop is that it checks for an sms... if it gets one it will check what the word in there is and carry out the action set... so far i just have start, stop, status1 and status2

if no text it received it will do the keepalive command but of course only if the heater has been given a command to start

and also it will check if there are a flame... if so it will send a text to say that the heater did light up

i will add some logic that send a text when the temp goes above say 60 degrees telling the car is ready to go

**Comment by boelle** [comment]
figured a few things and my code is now this: http://pastebin.com/KYRJXCbQ

have removed what seems to be dead variables in the beginning.... but any advice on cleaning it up more would be very welcome

**Comment by raldred** [comment]
@boelle thank you ever so much for sharing your code, i'm really very interested in this.
Did you build your own wbus interface board or use an off the shelf Arduino canbus shield like the one from sparkfun?
Would you mind sharing some pictures of the wiring or maybe a schematic 
Thanks

**Comment by stuartpittaway** [comment]
Hi Rob, my wiring diagram is on Github as well.
https://github.com/stuartpittaway/WebastoWBUSInterface

**Comment by boelle** [comment]
Hi

I will build my own, its not complete yet but it will be an eagle schematic thing

**Comment by raldred** [comment]
Ohh thanks @stuartpittaway that's great!

**Comment by boelle** [comment]
http://imgur.com/a/B2iMx

the black circle is the connection to the heater.. above is serial connection to GSM module

to the left is 2 dc/dc converter modules of the kind you find on ebay... a 15V suppresion diode to catch the most nasty spikes.. and a fat 22000uF cap to prevent to much drop when starting the engine

at the very top some leds to indicate status locally

not on there yet are some DS18B20 temp sensors for the cabin and outside

**Comment by boelle** [comment]
also missing in my code is a routine that checks if the gsm module lost connection and if so power cycle the gsm module

and the code i build this on only have TX line defined as far i can see... i wonder why as normally you would have a RX also

**Comment by boelle** [comment]
my code is based on this youtube video.... code is far down on the comments list..

but in the top ones i just got an reply that TX should be connected to RX/digi0... that is how i understand the authors reply

**Comment by raldred** [comment]
That's sounds really impressive. I arrived here from your comments on the youtube video :)
I'm actually only interested in heater control via my SIM900 i don't want/need anything else in the vehicle.

**Comment by boelle** [comment]
then we are going for the same goal... i just want the extra status leds so i dont have to send an sms all the time... i might add 2 buttons to turn it on/off locally

but some code that checks if the module is connected to the gsm network would be awesome... starting the engine has the risk of throwing the gsm off the network

**Comment by raldred** [comment]
@boelle I have dropped your code onto a gist maybe we could collaborate there rather than on pastebin. I've just tided up the formatting a bit for now.
https://gist.github.com/raldred/c9e9a592e6ed341af6c060fcf26cbe23

**Comment by boelle** [comment]
no problem

what i would be most interested in right now is a "void" that can check if there is connection to GSM network and if not power cycle the module

**Comment by raldred** [comment]
during the loop maybe add a check for `gsm.available()>0`

**Comment by boelle** [comment]
i use this lib: https://github.com/MarcoMartines/GSM-GPRS-GPS-Shield

they have some examples but they are pretty "thin"... and i actually asked if this was possible but no answer... 

but maybe its better to go for the standard gsm lib that comes with arduino?

can we do these "chats" on gist also?

**Comment by raldred** [comment]
Hi @boelle, Just wondered if you've made any progress?
I have now got my circuits built up, going to run some tests with the code.
I was just wondering why you're not initializing the wbus Serial1 during setup?
![image](https://cloud.githubusercontent.com/assets/37096/19628951/0e18f05a-9962-11e6-8606-f90093d1606f.png)

I have also been investigating using a HC-06 Bluetooth serial instead of the sim900

**Comment by boelle** [comment]
No not any progress as i ran out of money... not sure if i will get anything done before Xmas

Did not init wbus serial as the code i started out on did not do it... i have blindly copied off the code from the youtube video... maybe that was my mistake

http://pastebin.com/9jEW030e

and the guy also posted an alternative wbus.h

http://pastebin.com/8e8JQdX0

**Comment by boelle** [comment]
but i have come up with a PCB design

http://imgur.com/a/HsHLN

that is one to take care of the input side

-power
-DS18B20 for temps in car and outside car
-wbus chip instead of several components

there are pads on the edge that connects to the mega... but i have a few uno chips arround and maybe they can be used instead... but then the code would have to be changed to use softserial for both gsm and wbus and not sure if softserial can do that

i only miss the chip and board... if a bit lucky i will find money for that during next month. I also have to replace florescent tubes in an old monitor... its the continuous juggle with money when you are without a  job

**Comment by raldred** [comment]
Very nice.
Are you producing the IC chip yourself?
So far I have just built up the circuit on veroboard as per Stuart's design, as yet I have not tested, need to work on the code.

I'm think I'm going to use a Leonardo or uno. So I'll also need multiple software serial ports at least during testing so I can have the USB serial connected.
Some work will be needed to switch between the software serials because only one can listen at a time. It is possible though: https://www.arduino.cc/en/Tutorial/TwoPortReceive

Seems like it would be possible to do the GSM/Bluetooth Serial processing first.
Then switch to the wbus serial and listen on that, then repeat.

**Comment by boelle** [comment]
No the chip is a common one :-D

**Comment by raldred** [comment]
Ohh really, I've tried looking for it here, do you know it's code?

**Comment by boelle** [comment]
code?

you dont program it... 

datasheet: 

http://www.st.com/content/ccc/resource/technical/document/datasheet/4a/80/83/26/e0/78/4d/18/CD00000234.pdf/files/CD00000234.pdf/jcr:content/translations/en.CD00000234.pdf

**Comment by raldred** [comment]
I mean the IC package number. L9637
Thanks!

**Comment by boelle** [comment]
:-D

i could have build it with more resistors and transistors but i want to do it as clean i can... it takes me longer but i think it will be worth in the end

i thought of bluetooth but then again i would have limited range :-/

**Comment by raldred** [comment]
True, for me 30-50m is plenty.

**Comment by boelle** [comment]
but once i get the PCB's done i hope you can help me a bit with the code should i get in to trouble

if the code i came up with work i would be interested in a routine that could power cycle the GSM module if it loose connection to GSM network.

adding DS18B20 sensors should be easy enough

**Comment by raldred** [comment]
Great. I'll let you know how I get on.

**Comment by raldred** [comment]
No joy unfortunately. 
I've tried using your code @boelle as a base and nothing.
And I've reworked @stuartpittaway to remove the need for the screen and nothing there either. 

Can't get anything over the wbus, it just times out.

```
Keeping heater alive
wbus_init
wbus_init
wbus_init
Reading sensor failed
wbus error
Err 0xffff
```

I've had my meter on the circuit and it seems like the voltages are good and it's switching the transistors but it's not connecting to the car. (Land Rover Discovery 3)

**Comment by boelle** [comment]
hmmm....

since it's factory installed maybe its one of those heaters with canbus.... how and where do you connect? just guessing wild here

**Comment by raldred** [comment]
Its a definitely a webasto. You can buy an official webasto kit and it
works but it's €250. Maybe it's a different host address or something but I
have some software on my PC which can control it and read the status via an
odb2 Bluetooth thing. But it will only run the webasto when the main engine
is running.

**Comment by boelle** [comment]
how is the webasto connected to the car? and how do you connect to the webasto?

and some webasto are like you said programmed to only run as an supplemental heater and not as an aux heater

when you have enough daylight take some pictures and post them on imgur.com and post the link to the pictures here

**Comment by stuartpittaway** [comment]
Using the PC software, what does the heater report? Mine is an aux heater
from a range rover.

**Comment by raldred** [comment]
Thanks guys.
Ok I've had the Webasto Thermo Test software on it this morning using a usb odb2 cable which i've broken apart to connect up the wbus line to the heater.
Video: https://www.dropbox.com/s/7jts46tfab0ixuv/webasto_running.mp4?dl=0
I can control the heater with it completely no problem with the car is completely off
People on the Disco3 forum have made a device using a PIC controller using very similar code to what you have @boelle i wonder if one of them is a port of the other.

Maybe I just have an issue with my serial circuit

Here's the webasto device info screen:
It says it's an L5 SH
http://i.imgur.com/XewYwb8.jpg

Connection to heater:
I have cut the red wire (pin2) and connected my wbus line into it.
http://i.imgur.com/Sy1susk.png

**Comment by boelle** [comment]
hmm... my wbus is a single female spade plug at the fuses...

its a thermotop E if i remember correct... is pouring down outside but will try and get a pic when it stops

**Comment by raldred** [comment]
Mines fitted with a Thermo Top V

**Comment by boelle** [comment]
the workshop manual says that a Thermotop Z is the supplmental heater which only helps keeping things warm for those super efficient engines that cant keep them warm themselfes

ie it goes in standby as soon you start the engine and if coolant is below 60 C it will start... and i guess cutoff once it reaches 80 or so.

it can be converted to an aux heater with a retrofit kit if needed

**Comment by raldred** [comment]
I have a Thermo Top V
As the Webasto software can control with a simple USB serial converter surely I just need to emulate what the software's doing on the Arduino.
I'm getting a lot of my info from here: http://www.disco3.co.uk/wiki/Fuel_Burning_Heater

**Comment by boelle** [comment]
https://www.lrch.nl/sites/default/files/Disco_3_FBH_Workshop_Manual_Thermo_Top_V.pdf

in case you need it and did not have it

**Comment by boelle** [comment]
"The Thermo Top V auxiliary water heater is used to compensate
the heat deficit on consumption-optimised vehicle
engines and can be upgraded to an additional/auxiliary
heater with an upgrade kit."

so no need to convert it.... i think

**Comment by boelle** [comment]
how my W bus connection looks like

http://imgur.com/3nYBUGa

it can also be done at the clock connector inside the car but dont have the pinout handy

**Comment by raldred** [comment]
Yeh they sold a kit to russian markets only.

According to people on the disco3 website you can retrofit park heater functionality easily.
http://www.disco3.co.uk/wiki/Fuel_Burning_Heater#Park_Heater_Functionality

I was specifically looking at this post:
http://www.disco3.co.uk/forum/topic106191.html
Where a user has used a PIC controller which looks like it implements the wbus functions almost exactly as you have for your sim900 arduino version.

Here's the pic controller code:
https://github.com/raldred/webasto_pic_fbh

**Comment by raldred** [comment]
I'm almost certain now that my issue lies with my vero board circuit.

**Comment by boelle** [comment]
Let's hope for that or else it sounds like a nightmare :-/

**Comment by boelle** [comment]
i found one of those bluetooth to obd2 dongles... to unreliable and bluetooth range (10 meter) is not quite enough for me.

if i could connect i could just use the thermotest software on my laptop inside.... of course that would only work when i'm home so still interested in something with grater range (gsm....)

**Comment by boelle** [comment]
I'm very Tempted to look more at this one:

https://store.open-electronics.org/GSM-GPRS-GPS-modem-ft960k

remove the usb part from it and connect it to a obd2 to serial dongle

then another of the gsm modems at the other end... make a data call to the car and use the thermostop software....

just thoughts thou

**Comment by stuartpittaway** [comment]
My heater is a thermotop V and I had to connect the wbus to the middle
multiplug which has 8 connection on it.

**Comment by raldred** [comment]
Pin2 @stuartpittaway ? Mine has a read wire which was removed on later models.
![Webasto plug](http://www.disco3.co.uk/gallery/albums/userpics/13040/wiki1_FBH%20connector.JPG)
I have made up your circuit. I'm having trouble diagnosing what's wrong.
- I don't suppose you have any pictures of yours wired up, where you're getting power from, grounding?
- Does my arduino RX go to the TTL converter TX?
- I assume I only need a single GND as both sides are linked.

@boelle I wanted to avoid using one of those things too hence the Arduino. I can get a better range using a HC06 and I can't be bothered messing with GSM :)
I have a DFrobot SIM908 shield sat here doing nothing if you're interested?

![Sim908](https://www.dropbox.com/s/0585z0jjdaecb7v/2016-11-06%2016.58.20.jpg?dl=1)

**Comment by stuartpittaway** [comment]
Use the interface you have built from the PC and the
WebastoHeaterWBusArduinoInterface software for testing.  Just use the USB
to ttl adapter to connect the interface to the PC

**Comment by boelle** [comment]
from usb/ttl adapter you need RX, TX and GND to the interface

from the heater you need Wbus and GND.

Both GNDs need to connect to each other.... if not the voltage levels are not correct

**Comment by boelle** [comment]
i guess there will be some data even if you leave the arduino out.

if that is the case we know it works.

@stuartpittaway am i right in my assumptions?

**Comment by raldred** [comment]
Doh I don't have a USB/TTL adapter. This skips the arduino right for debugging right?
Ok Yeh I did have GND connected on the 6 pin side to arduino and the 2pin side to the car.

**Comment by boelle** [comment]
@raldred going for dinner now, but where are you located and how much do you want for  the board?

**Comment by raldred** [comment]
No worries, thanks for all your help guys, gonna order me a usb/ttl adapter.

I'm in the UK, I can probably ship it to denmark easily how does $50 sound? cost me quite a bit more, but it's sat here in a drawer doing nothing.
It's brand new never used. https://www.dfrobot.com/index.php?route=product/product&filter_name=gps&product_id=673

**Comment by stuartpittaway** [comment]
Don't you use a ttl adapter for programming the Arduino?

**Comment by raldred** [comment]
No I'm using a Leonardo which runs an ATmega32u4 it has a built in USB interface.

**Comment by boelle** [comment]
@raldred $50 sounds ok to me, but can we wait until after New years?

want to treat my girlfriend with a pair of headphones and graphics card (she is a Gamer...). 

Will also give me time to figure how i can make a serial data connection over the gsm network.

the bucket of gold at the end of the rainbow would be if the test app was also made for phones... but i guess that would be pushing it

**Comment by raldred** [comment]
Of course. No panic. Just let me know whenever.
Btw I've soldered my npn transistors backwards. Doh!  Resoldering now
hopefully it will just work now. Ha.

**Comment by boelle** [comment]
oh well you are not the only, first or last one that have done that :-D

**Comment by boelle** [comment]
@stuartpittaway going through my box of random parts. Should there be a problem replacing BC548 with BC547B ?

**Comment by stuartpittaway** [comment]
It shouldn't matter, as long as you replace npn with npn or pnp with pnp

**Comment by raldred** [comment]
Good news, It's working... partially. TX is working so I can send on/off commands to the heater, the LED on the interface is flashing as commands send and it fires up and shuts down the heater but I'm not getting anything back on RX.

**Comment by stuartpittaway** [comment]
Might still be an issue with the direction of the transistors

**Comment by raldred** [comment]
Yeh, I've checked them all again and i'm pretty sure they're correct. 
Is there a particular transistor that's involved with RX? I wonder if i've damaged it desoldering
![circuit](https://www.dropbox.com/s/78bezfh2d48y6fz/2016-11-06%2022.02.23.jpg?dl=1)

**Comment by boelle** [comment]
https://github.com/stuartpittaway/WebastoWBUSInterface/blob/master/Webasto%20WBus%20Arduino%20Serial%20Interface%20mk7_schem.png

i THINK its the one in the lower left corner

**Comment by boelle** [comment]
@stuartpittaway or maybe he has not connected it right to the heater? 2 wbus connections at the right might have caused confusion

does the coloured lines mean anything special?

**Comment by boelle** [comment]
https://raw.githubusercontent.com/stuartpittaway/WebastoWBUSInterface/master/Webasto%20WBus%20Arduino%20Serial%20Interface%20mk7_bb.png

looks like everything is mirrored?

**Comment by raldred** [comment]
Yeh because fritzing shows components on the same side as the copper on
vero boards.

**Comment by boelle** [comment]
good to know.. i have found the transistors and the led, but did not have enough of the resistors so will have to visit the electronic shop and get more tomorrow at some point.

**Comment by boelle** [comment]
btw... what range do you get with the bluetooth? just curious

**Comment by stuartpittaway** [comment]
Mine is a mirror image of your circuit! The flats on the transistors point
to the 3 pin header, not the 6 pin!!

**Comment by raldred** [comment]
Ahh I dunno maybe i've messed up weird that TX would work now. I'm doing to build a fresh circuit I think. Don't suppose you could share some pictures of both faces of your vero board @stuartpittaway ?

**Comment by raldred** [comment]
Good news I found a tiny fault on my ground rail, repaired and I'm getting responses back from the heater.
However I'm not sure how to interpret some of the numbers.

The OP_STATE codes are coming back fine.
But the temperature and voltage readings and other sensor readings don't make much sense.

When it's off (and cold) I get the following readouts

```
t210 v0
f0 h0 g0
```

temp 210?

```
heater on
t252 v2816
f0 h0 g0
```

when it's been running a few seconds

```
t253 v15616
f0 h0 g0
```

once the burner is running and it's heating

```
t206 v28928
f0 h0 g0
```

the temp has gone down (it's definitely going up)
flame,heat energy and glow values still read 0

**Comment by boelle** [comment]
hmmm... i bet that the temp number is offset so that you have to subtract something to get it right

but not sure.... i'm still on project 3d printer rebuild. but i got half the parts for the interface. might get a change to go to electronics store next week

any chance you could throw some photos of the board to http://imgur.com   ?

**Comment by raldred** [comment]
Yeh the header file suggests it's offset by 50degrees so 20 would actually read as 70 but 210 seems wildly out. I would also expect the temp to increase as it heats but it stays fairly stable.

Sure, it's a bit of mess now as I had to desolder bits and mess it fix a few broken tracks.
http://imgur.com/a/XhxY9

**Comment by boelle** [comment]
np... at least i will have something to start on

but yes the temp thing seems odd.. maybe @stuartpittaway can shine some light on it

**Comment by boelle** [comment]
hmm.. maybe @stuartpittaway can shine some light on the temp thing

but you are able to drive it with the thermotest software?

https://raw.githubusercontent.com/stuartpittaway/WebastoWBUSInterface/master/Webasto%20WBus%20Arduino%20Serial%20Interface%20mk7_schem.png

i wonder what the different colours on the schematics means and why there are 2 wbus connectors

**Comment by raldred** [comment]
I think the two wbus are just separated on the schematic for readability.
they're actually both connected to the same single output on the 3 pin
connector on the 12v side. I was going to test the thermo test with it
later.

**Comment by stuartpittaway** [comment]
I used a 3 pin header for 12v, 0v and wbus.

Stu.

**Comment by boelle** [comment]
@stuartpittaway can you shine some light on the text output that @raldred showed above?

**Comment by stuartpittaway** [comment]
Not really! Ive never got my heater to actually work 😎

Can you get the reading from the Webasto software instead?

**Comment by boelle** [comment]
He will test it later today if not mistaken. Me will be the slow one :-/

**Comment by raldred** [comment]
Yeh mrs is out with the car, I'll test with the webasto software when she's back.
That's unfortunate @stuartpittaway did you have a faulty heater or just ran out of patience?

**Comment by boelle** [comment]
@raldred have you tested the range with the bluetooth module you have? i read that they only give 10 meters of range

**Comment by raldred** [comment]
Yeh i was very surprised, It probably depends, interference and line of sight etc. But I walked about ~30 meters up the road. It was stable and I could still communicate with the arduino. It would probably work further.

**Comment by boelle** [comment]
i cut a bit of strip board to size and used a black marker to indicate where i think you did the cuts

Did i get it right?

http://imgur.com/a/61Shn

**Comment by raldred** [comment]
Yeh that looks the same as mine, although @stuartpittaway said its actually mirrored.
So the components are actually on the back side

**Comment by boelle** [comment]
ahh yes: https://raw.githubusercontent.com/stuartpittaway/WebastoWBUSInterface/master/Webasto%20WBus%20Arduino%20Serial%20Interface%20mk7_bb.png

looks the same if i let the tracks point up. I can work with inserting the components from the other side. It takes a tad more to confuse me... but not much more :-D

**Comment by boelle** [comment]
forgot to ask. did you do a test with the thermotest software?

**Comment by stuartpittaway** [comment]
That image is a top down view, so the components side of the strip board!

**Comment by boelle** [comment]
must have gotten it right this time

http://imgur.com/a/MePz2

**Comment by stuartpittaway** [comment]
Looks good. I tried to make the interface board as small as possible on
strip board. Think I managed it, but does make the building more difficult.

On 14 Nov 2016 16:24, "Bo Herrmannsen" notifications@github.com wrote:

must have gotten it right this time

http://imgur.com/a/MePz2

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub
https://github.com/stuartpittaway/WebastoHeaterWBusArduinoInterface/issues/1#issuecomment-260382403,
or mute the thread

**Comment by boelle** [comment]
hehe, it just that fritzing image that makes it a bit "odd"

lucky others can use our pics to get it right too

**Comment by stuartpittaway** [comment]
I'll try and upload some more images later

**Comment by poeml** [comment]
I'm looking forward to, lurking since the beginning of this thread and following closely!

Thanks for sharing!!

Peter

**Comment by raldred** [comment]
@boelle sorry I haven't had time to check the thermo test will try later.

**Comment by boelle** [comment]
No rush

I found a 5V FTDI cable so initial i can skip the resistor dividder

And i got hold off 2-3 meters of 3 wire cable so i can feed the Wbus inside the cabin. 

Starting to debate with myself if bluetooth or Xbee cant do the trick instead of GSM

If the test software was made as an android app it would make the debate short :-D

**Comment by raldred** [comment]
This is the bluetooth app im using to test
https://play.google.com/store/apps/details?id=Qwerty.BluetoothTerminal

You just pair your phone with the HC-06
Load up the app, connect to the HC-06 and boom
Send it text strings

**Comment by boelle** [comment]
hehe... i was more thinking like an app that does the same as the thermotest software. 

But i guess if i where to make a module to go with the phone i would weld in memory what string to start it

**Comment by raldred** [comment]
Ahh sorry yes well maybe that's where we can get to eventually :D

**Comment by boelle** [comment]
it can be done:

http://www.instructables.com/id/Arduin-Adroid-USB-Serial-Communication/

i was satisfied with just the serial part

oh well... this is not the time to think to far

**Comment by bmwfarmer** [comment]
Great project.
Please share U8glib and M2tk version and download link. Your .ino doesn't verify with newest ones.
I can start my Webasto with A6 GSM module and Arduino Uno by making unanswered call to it.
But it would be nice to have a timer and lcd.
Cheers

**Comment by stuartpittaway** [comment]
As its opensource, feel free to make the necessary changes and push into
Github :-)

**Comment by raldred** [comment]
Hey guys, sorry for the radio silence, I've snowed under with work and have only this week had chance to get out there with the arduino again.
Happy to report the interface I built from stuarts fritzing works perfectly with the webasto software over a USB serial, despite be building a mirrored circuit 👍 

And after a little debugging I found a problem with my version of the code on the arduino.
So i have a fully functional Bluetooth controller, I just need to tidy up the wiring, box it up and bury it behind the dashboard.

Thanks @stuartpittaway and @boelle you've been invaluable support.
I'll create a PR with my code for using a BT serial (im not using an LCD, all the status reporting is going back to the device connected to the BT serial) and I hope to get a blog post or something up showing the whole setup for others to use, probably sometime over the holidays.

@bmwfarmer the repo is 100% open source you're more than welcome to make changes to improve support.

**Comment by boelle** [comment]
@raldred can you lift the blanket a bit covering your setup and say what the fault in the code was?

for myself i have not had time to go get the remaining parts yet but what i have is soldered in and i digged out a 5v FTDI cable so i can test it as soon i get my act together. I have even figured that i will be doing BT also. It will be so rare that we will need to fire it up remotely that is will not be worth adding GSM. at least not until summer sometime.

**Comment by raldred** [comment]
@boelle course. It was just a stupid error on my part, took me a while to see it. I've stripped out the parts of Stuarts code that I needed in doing so I had messed up the calls for QUERY_STATE and QUERY_SENSORS in a similar way to Stuart I have it cycling between reporting the state and the sensor data. 

During the loop I was switching the reporting information to the serial but not actually switching the wbus query command so it was only every calling QUERY_STATE this caused the output to look corrupt because the data coming back was actually state info not sensor info.

Great, yeh like I me mentioned the other week, I've had really good results with the BT range. Once I've packed it all up and have it installed properly in the car I'll do some more testing.

**Comment by boelle** [comment]
think i will go with a pair of these 

http://www.ebay.co.uk/itm/Sena-Technologies-Parani-BCD100-Bluetooth-Class-1-OEM-module-4-simultaneous-/291806135983?hash=item43f10046af:g:hj8AAOSwbYZXc5Nl

**Comment by raldred** [comment]
I'm using this
http://www.ebay.co.uk/itm/171941965804

**Comment by boelle** [comment]
hehe, those are only good for about 10 meters, give/take a little.

i need at least 3 times that. but i will start working on the bluetooth part after xmas

**Comment by stuartpittaway** [comment]
I've recently updated GITHUB with a re-compile of the Webasto heater
software running under the platform.io - it includes the latest version of
the support libraries.

**Comment by raldred** [comment]
You say that but like I mentioned earlier I had no problem keeping a stable
connection ~50meters

**Comment by boelle** [comment]
Oh, that helps a lot

anyways if they are not enough i can always use them for something else

also the lower the BAUD speed the longer they should reach. so says people but not sure if that is the truth

**Comment by stuartpittaway** [comment]
You may all find this very useful...
http://byron76.blogspot.co.uk/2011/09/upgrade-your-bluetooth-module.html

I've used it a few times to give better firmware for the bluetooth modules.

**Comment by boelle** [comment]
hehe

i know that some boards are sold as 2 versions where only the firmware limits the transmit power

is this the case here? the instructions does not says much

**Comment by bmwfarmer** [comment]
Webasto K-line is one wire interface, so your k-line adapter should bounce everything back to your rx pin. This must happen even it is not connected to Webasto, 12v is still needed for testing.

**Comment by raldred** [comment]
@bmwfarmer what are your referring to exactly? Not really sure what the wbus interface has to do a bluetooth module.

**Comment by boelle** [comment]
I finally came arround to getting those last resistors and finish up the board:

http://imgur.com/a/T8gUb

not on pictures are a 3 wire cable with the right plug crimped on. I also got a ftdi cable with 5v TTL label on it so i should not need a level converter.

next step would be to get one of those lone range bluetooth modules. I have allready tested that the one build in my laptop is not up to the task as my garage 30 meters away and i live on 1st floor which adds a bit on top of that. normal bluetooth is speced at 10 meters so yeah i will need something else no matter what angle you look at

**Comment by boelle** [comment]
i might make some eagle files out of this so people that should stumble across this can order a board from say oshpark or whatever they pick

**Comment by stuartpittaway** [comment]
Great work!

**Comment by boelle** [comment]
@stuartpittaway if i make those files would you include them if i made a PR and submitted it?

i know some might screem that eagle is not free but it is as long you dont have to go over 100*80 which i think would be the case here

**Comment by stuartpittaway** [comment]
Sure happy to host files

**Comment by caspervanzoest** [comment]
Hi all,

Since there is a lot of talking about the circuit from the other branch going on here,
I would like to ask a few questions about the usage on this thread.

Please see my issue on the circuit branch:
[https://github.com/stuartpittaway/WebastoWBUSInterface/issues/1](https://github.com/stuartpittaway/WebastoWBUSInterface/issues/1)

Thanks in advance!

**Comment by CsepregiArtur** [comment]
Hello

Whit this Wbus comunication is more easier to control and read data from webasto?

Thank you
